### PR TITLE
server-streaming: release reference on request after method call

### PR DESCRIPTION
**Pull request by Gordiychuk**
fixing https://github.com/grpc/grpc-java/issues/5636

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @Gordiychuk isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by Gordiychuk** [comment]
Signed

File: stub/src/main/java/io/grpc/stub/ServerCalls.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ private ServerCalls() {
       private final ServerCall<ReqT, RespT> call;
       private final ServerCallStreamObserverImpl<ReqT, RespT> responseObserver;
       private boolean canInvoke = true;
+      private boolean messageReceived = false;
```
Comment (carl-mastrangelo) [inline stub/src/main/java/io/grpc/stub/ServerCalls.java]: optional: drop the `= false`, as it's the default.

File: stub/src/main/java/io/grpc/stub/ServerCalls.java [hunk 2]
```diff
@@ -169,6 +171,7 @@ public void onHalfClose() {
         }
 
         method.invoke(request, responseObserver);
+        request = null;
```
Comment (ejona86) [inline stub/src/main/java/io/grpc/stub/ServerCalls.java]: It looks like you only need to add this line and not bother with the others. It doesn't seem `messageReceived` is improving any behavior.

**Comment by ejona86** [comment]
This was a good find. Thank you!

File: stub/src/main/java/io/grpc/stub/ServerCalls.java (hunk 2 above)
Comment (Gordiychuk) [reply stub/src/main/java/io/grpc/stub/ServerCalls.java]: done

**Comment by ejona86** [comment]
@Gordiychuk, thank you!

**Pull request merged**